### ISSUE TITLE
Release 1.17

### ISSUE THREAD

**Issue by laurent182** [civetweb/civetweb#1253]
Hi,

I'd like to know if version 1.17 will be released soon.
I would need to specify a second path with the fallback_document_root parameter.

Regards,

**Comment by bel2125** [comment]
There are some issues with continuous integration, and the experts on that are not available.
I am confident it is not an issue with the code but with some build script - probably I just have to drop them from the project, since I have the impression they are consuming time instead of helping to save time. I will have a look after Easter.

**Comment by DL6ER** [comment]
I'd like to bring this question into focus again. The last release was more than 1 year ago and it seems CI issues might have been fixed meanwhile?

We are embedding CivetWeb in [Pi-hole](https://www.pi-hole.net) and would like to benefit from the changes meanwhile, especialy the onces directly connected to our project:
- https://github.com/civetweb/civetweb/pull/1208
- https://github.com/civetweb/civetweb/pull/1209
- https://github.com/civetweb/civetweb/pull/1277
- https://github.com/civetweb/civetweb/pull/1278
- https://github.com/civetweb/civetweb/pull/1279

**Comment by bel2125** [comment]
CI/CD has not been fixed.

CivetWeb is not commercial. Since a while TravisCI asks for a paid plan (see screenshot). 
They gave some free credits, but we usually use them up very quickly.
![image](https://github.com/user-attachments/assets/8e143037-8c57-4aa0-90fd-6173f1a438a2)

AppVeyor, originally only used for Windows, now also offers Linux support (already since a while).
This works - except the Visual Studio 2010 and 2012 build that could declared as "end of support" by now. 
![image](https://github.com/user-attachments/assets/a8288805-d4f0-44e2-9838-0bdc0a6b3ffd)

I have no build/test infrastructure for MacOS. We can only cancel MacOS support ... basically: If it works, fine, if it does not and you can provide a fix, I can merge it, if you cannot provide a fix you are out of luck.

**Comment by bel2125** [comment]
I am not an expert on this CI/CD systems.
In fact a contributor made most of it, who is not around anymore.
I know this issue with CI/CD will not fix itself, but I have a hard time doing it - I understand that delaying the release because of this is also not really good.

**Comment by DL6ER** [comment]
There is free CI in GitHub ([Actions](https://github.com/features/actions)) including MacOS for public repositories. The transition may, of course, take a lot of time but often most of what is there can be reused 1:1. Just mentioning this here in case you have time and haven't heard much about it. The Pi-hole team, for instance, used Travis, then CircleCI and now - for quite some time actually - said **free Actions for building and testing binaries for various architectures such as x86, ARM and even RISCV**. See [here](https://github.com/pi-hole/FTL/tree/master/.github/workflows) in case you're interested. The most recent code is actually [here](https://github.com/pi-hole/FTL/tree/development-v6/.github/workflows) but this link will stop working after the next release (when the branch goes away).

**Comment by bel2125** [comment]
Yes, we definitely should switch from Travis to something else. I must admit Travis gave me some free credits a couple of times when I asked, but they were used up quickly. Actually the Travis platform works fine and I do understand they need to make some money. But CivetWeb is completely free - nobody has to pay for it, not even donate - but also we don't pay for anything. 
I know about GitHub actions, but have not used them before. In fact all of the build automation has been done by someone else - I have never dealt with this topic myself and cannot do much more than small modifications to existing scripts with a lot of try and error.
Probably I put a request for help somewhere.

**Comment by yubiuser** [comment]
> Probably I put a request for help somewhere.

I started working on this, see https://github.com/civetweb/civetweb/pull/1284. Linux build has been added already, currently working on MacOS. Afterwards, I'll try to port the unit test/coverage as well.

Successful linux run can bee seen at https://github.com/yubiuser/civetweb/actions/runs/10439171287

**Comment by yubiuser** [comment]
@bel2125 

Travis CI has been ported to Github Action CI which is working fine. Do you think it's time for 1.17 release?
